tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedThreadFactory/injected/ManagedThreadFactoryDefinitionInjectedServlet.java

```diff
@@ -0,0 +1,235 @@
+                throw new AssertionError().initCause((Throwable) result);
+            assertTrue(result instanceof ContextService, "Application context must be propagated to first thread "
+                    + "per java:comp/concurrent/ThreadFactoryInjB configuration.");
+
+            result = txTaskResult.get(MAX_WAIT_SECONDS, TimeUnit.SECONDS);
+            if (result instanceof Throwable throwable) {
+                throw new AssertionError().initCause(throwable);
+            }
+            assertEquals(result, Status.STATUS_NO_TRANSACTION,
+                    "Transaction context must be cleared from async Callable task "
+                    + "per java:comp/concurrent/ThreadFactoryInjB configuration.");
+        } finally {
+            IntContext.set(0);
+            blocker.countDown();
+        }
+    }
+
+    /**
+     * ManagedThreadFactory can be supplied to a ForkJoinPool, causing ForkJoinPool
+     * tasks to run with the thread context and priority as configured.
+     */
+    public void testParallelStreamBackedByManagedThreadFactory() throws Throwable {
+        ForkJoinPool fj = null;
+        try {
+            IntContext.set(1000);
+            StringContext.set("testParallelStreamBackedByManagedThreadFactory-1");
+
+            IntContext.set(2000);
+            StringContext.set("testParallelStreamBackedByManagedThreadFactory-2");
+
+            fj = new ForkJoinPool(4, threadFactoryInjA, null, false);
+
+            IntContext.set(3000);
+            StringContext.set("testParallelStreamBackedByManagedThreadFactory-3");
+
+            ForkJoinTask<Optional<Integer>> task = fj.submit(() -> {
+                return Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9).parallelStream().map(num -> {
+                    assertEquals(StringContext.get(), "",
+                            "Third-party context type StringContext must be cleared on " + "ForkJoin thread.");
+                    try {
+                        assertNotNull(InitialContext.doLookup("java:app/concurrent/ContextA"),
+                                "Application context must be propagated to ForkJoin thread");
+                    } catch (NamingException x) {
+                        throw new CompletionException(x);
+                    }
+                    return num * Thread.currentThread().getPriority() + IntContext.get();
+                }).reduce(Integer::sum);
+            });
+
+            Optional<Integer> result = task.join();
+            assertEquals(result.get(), Integer.valueOf(9180),
+                    "Third-party context type IntContext must propagated to ForkJoin threads "
+                            + "(thousands digit should be 9) and thread priority (4) must be enforced "
+                            + "on ForkJoin threads (hundreds/tens/ones digits must be 4x5x9=180) "
+                            + "per configuration of the ManagedThreadFactoryDefinition and ContextServiceDefinition.");
```

Comment:
This has a copy/paste error from the other test that had IntContext=1000 on the thread when the lookup was performed.  For this one, the IntContext is 0 when looked up, and needs to be updated as follows:

```suggestion
            IntContext.set(2000);
            StringContext.set("testParallelStreamBackedByManagedThreadFactory-2");

            fj = new ForkJoinPool(4, threadFactoryInjA, null, false);

            IntContext.set(3000);
            StringContext.set("testParallelStreamBackedByManagedThreadFactory-3");

            ForkJoinTask<Optional<Integer>> task = fj.submit(() -> {
                return Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9).parallelStream().map(num -> {
                    assertEquals(StringContext.get(), "",
                            "Third-party context type StringContext must be cleared on " + "ForkJoin thread.");
                    try {
                        assertNotNull(InitialContext.doLookup("java:app/concurrent/ContextA"),
                                "Application context must be propagated to ForkJoin thread");
                    } catch (NamingException x) {
                        throw new CompletionException(x);
                    }
                    return num * Thread.currentThread().getPriority() + IntContext.get();
                }).reduce(Integer::sum);
            });

            Optional<Integer> result = task.join();
            assertEquals(result.get(), Integer.valueOf(180),
                    "Third-party context type IntContext must be propagated to ForkJoin threads "
                            + "(thousands digit should be 0) and thread priority (4) must be enforced "
                            + "on ForkJoin threads (hundreds/tens/ones digits must be 4x5x9=180) "
                            + "per configuration of the ManagedThreadFactoryDefinition and ContextServiceDefinition.");
```